src/View/AbstractDirectoryView.vala

```diff
@@ -2931,6 +2930,8 @@ namespace FM {
                 if (slot.directory.is_local || NetworkMonitor.get_default ().get_network_available ()) {
                     /* cannot get file info while network disconnected. */
                     GOF.File? target_file;
+                    GOF.File? file = path != null ? model.file_for_path (path) : null;
```

Comment:
This line was moved here so the function is only called when required instead of at every event.